tests/SprykerEcoTest/Zed/PunchoutCatalogs/Business/PunchoutCatalogsFacadeTest.php

```diff
@@ -0,0 +1,196 @@
+
+        $updatedPunchoutCatalogConnectionTransfer = $this->getFacade()
+            ->findConnectionById($punchoutCatalogConnectionTransfer->getIdPunchoutCatalogConnection());
+
+        $this->assertEquals($punchoutCatalogConnectionTransfer->getUsername(), $updatedPunchoutCatalogConnectionTransfer->getUsername());
+    }
+
+    /**
+     * @return void
+     */
+    public function testUpdateConnectionUnsuccessfulIfNoConnectionExist(): void
+    {
+        // Arrange
+        $punchoutCatalogConnectionTransfer = new PunchoutCatalogConnectionTransfer();
+        $punchoutCatalogConnectionTransfer->setIdPunchoutCatalogConnection(static::NOT_EXISTING_CONNECTION_ID);
+
+        // Act
+        $punchoutCatalogResponseTransfer = $this->getFacade()
+            ->updateConnection($punchoutCatalogConnectionTransfer);
+
+        // Assert
+        $this->assertFalse($punchoutCatalogResponseTransfer->getIsSuccessful());
+    }
+
+    /**
+     * @return int
+     */
+    protected function createPunchoutCatalogsConnection(): int
+    {
+        $companyTransferBusinessUnit = $this->createCompanyBusinessUnit();
+
+        $punchoutCatalogConnectionEntity = (new PgwPunchoutCatalogConnection())
+            ->setName(static::CONNECTION_NAME)
+            ->setUsername(static::CONNECTION_USERNAME)
+            ->setType(static::CONNECTION_TYPE)
+            ->setFormat(static::CONNECTION_FORMAT)
+            ->setFkCompanyBusinessUnit($companyTransferBusinessUnit->getIdCompanyBusinessUnit());
+
+        $punchoutCatalogConnectionEntity->save();
+
+        return $punchoutCatalogConnectionEntity->getIdPunchoutCatalogConnection();
+    }
+
+    /**
+     * @return \Generated\Shared\Transfer\CompanyTransfer
+     */
+    public function createCompany(): CompanyTransfer
+    {
+        return $this->tester->haveCompany(
+            [
+                CompanyTransfer::NAME => 'Test company',
+                CompanyTransfer::STATUS => 'approved',
+                CompanyTransfer::IS_ACTIVE => true,
+                CompanyTransfer::INITIAL_USER_TRANSFER => new CompanyUserTransfer(),
+            ]
+        );
+    }
+
+    /**
+     * @return \Generated\Shared\Transfer\CompanyBusinessUnitTransfer
+     */
+    public function createCompanyBusinessUnit(): CompanyBusinessUnitTransfer
+    {
+        return $this->tester->haveCompanyBusinessUnit(
+            [
+                CompanyBusinessUnitTransfer::NAME => 'test business unit',
+                CompanyBusinessUnitTransfer::EMAIL => 'test@spryker.com',
+                CompanyBusinessUnitTransfer::PHONE => '1234567890',
+                CompanyBusinessUnitTransfer::FK_COMPANY => $this->createCompany()->getIdCompany(),
+            ]
+        );
+    }
+
+    /**
+     * @return \SprykerEco\Zed\PunchoutCatalogs\Business\PunchoutCatalogsFacadeInterface
+     */
+    protected function getFacade(): PunchoutCatalogsFacadeInterface
```

Comment:
useless method. use `$this->tester->getFacade()` directly where it is needed.

Reply:
`$this->tester->getFacade()` returns `\Spryker\Zed\Kernel\Business\AbstractFacade`
So `getFacade()` method provides code highlighting of IDE, so I would prefer to left it as is

Reply:
Add to tester's doc block:
```
* @method \SprykerEco\Zed\PunchoutCatalogs\Business\PunchoutCatalogsFacadeInterface getFacade()
```